packages/compiler/core/decorator-utils.ts

```diff
@@ -432,14 +432,32 @@ export function validateDecoratorUniqueOnNode(
   return true;
 }
 
+/**
+ * Validate that a given decorator is not on a type or any of its base types.
+ * Useful to check for decorator usage that conflicts with another decorator.
+ * @param context Decorator context
+ * @param type The type to check
+ * @param badDecorator The decorator we don't want present
+ * @param givenDecorator The decorator that is the reason why we don't want the bad decorator present
+ * @param includeHeritage Whether to check base types for the bad decorator too
+ * @returns Whether the decorator application is valid
+ */
 export function validateDecoratorNotOnNode(
```

Comment:
I guess if this method is the one checking for the heritage does it still make sense to be called like that?
The goal of the other one validateDecoratorOnNode was to emit an error if the same decortor was used twice on teh same node but would not complain if you used it twice via `op is` or `model is`